cdap-data-fabric/src/main/java/co/cask/cdap/data/runtime/DataSetServiceModules.java

```diff
@@ -90,6 +94,14 @@ protected void configure() {
 
             bind(DatasetOpExecutor.class).to(LocalDatasetOpExecutor.class);
             expose(DatasetOpExecutor.class);
+
+            bind(DatasetTypeService.class)
+              .annotatedWith(Names.named(BASE_DATASET_TYPE_SERVICE))
+              .to(DefaultDatasetTypeService.class)
+              .in(Scopes.SINGLETON);
```

Comment:
We didn't have singleton scope before, right? Any reason for adding it here and in other methods in this class?